### ISSUE TITLE
Player crashes randomly in Godot 4.2 on macOS

### ISSUE THREAD

**Issue by denishacquin** [godotengine/godot#86788]
### Tested versions

- Reproducible in 4.2.1 stable

### System information

Godot v4.2.1.stable - macOS 14.1.0 - Vulkan (Forward+) - integrated Apple M3 Max - Apple M3 Max (14 Threads)

### Issue description

The editor player crashes randomly on this simple project. Sometimes I have no logs at all, sometimes I have the following error with no backtrace at all, and once I managed to have the backtrace printed.
EDIT: also crashes when running export build on macOS.

```
  ================================================================
  handle_crash: Program crashed with signal 11
  Engine version: Godot Engine v4.2.1.stable.official (b09f793f564a6c95dc76acc654b390e68441bd01)
  Dumping the backtrace. Please include this when reporting the bug to the project developer.
  [1] 1   libsystem_platform.dylib            0x0000000180e37a24 _sigtramp + 56
  [2] 2   ???                                 0xffff800180e840a8 0x0 + 18446603342678868136
  [3] 3   CoreFoundation                      0x0000000180e840a8 __CFBasicHashDrain + 356
  [4] 4   CoreFoundation                      0x0000000180fb7e28 _CFRelease + 292
  [5] 5   CoreFoundation                      0x0000000180eb82a4 __CFBinaryPlistWriteOrPresize + 308
  [6] 6   Foundation                          0x0000000181fda6e8 -[NSKeyedArchiver finishEncoding] + 640
  [7] 7   AppKit                              0x000000018481e634 __62-[NSPersistentUIArchiver finishEncodingWithCompletionHandler:]_block_invoke + 64
  [8] 8   AppKit                              0x000000018541cbb8 ___NSPersistentUIDispatchQueueAsync_block_invoke + 76
  [9] 9   libdispatch.dylib                   0x0000000180c59cb8 _dispatch_call_block_and_release + 32
  [10] 10  libdispatch.dylib                   0x0000000180c5b910 _dispatch_client_callout + 20
  [11] 11  libdispatch.dylib                   0x0000000180c62ea4 _dispatch_lane_serial_drain + 748
  [12] 12  libdispatch.dylib                   0x0000000180c639d4 _dispatch_lane_invoke + 380
  [13] 13  libdispatch.dylib                   0x0000000180c6e61c _dispatch_root_queue_drain_deferred_wlh + 288
  [14] 14  libdispatch.dylib                   0x0000000180c6de90 _dispatch_workloop_worker_thread + 404
  [15] 15  libsystem_pthread.dylib             0x0000000180e05114 _pthread_wqthread + 288
```

### Steps to reproduce

Run the main scene and move around with WASD.
Sometimes it will crash within a second, sometimes after a while.

### Minimal reproduction project (MRP)

[DemoProject.zip](https://github.com/godotengine/godot/files/13831974/DemoProject.zip)

**Comment by ajreckof** [comment]
When doing a MRP you need to remove all things not essential to the problem. Next time please try and remove as much things as possible to facilitate the job of contributors. here is a MRP (just open the MRP and launch the main scene it should automatically crash ) : 
[MRP.zip](https://github.com/godotengine/godot/files/13837846/MRP.zip)
I've found the exact problem and I'm looking into a solution

**Comment by denishacquin** [comment]
I actually did try to remove as much stuff as possible - apparently not enough!

Thanks for looking into this :)

**Comment by AThousandShips** [comment]
Can someone confirm if this has been fixed by:
* https://github.com/godotengine/godot/pull/96856

**Comment by ajreckof** [comment]
I can confirm this is fixed by #96856 (tested on master)
